lib/pbench/test/unit/server/database/test_datasets_metadata_db.py

```diff
@@ -111,6 +111,53 @@ def test_metadata_remove(self, db_session, create_user):
         assert ds.metadatas == []
 
 
+class TestInternalMetadata:
+    def test_dataset_full(self, provide_metadata):
+        ds = Dataset.query(name="drb")
+        metadata = Metadata.getvalue(ds, "dataset")
+        assert metadata == {
+            "access": "private",
+            "created": "2020-02-15T00:00:00+00:00",
+            "name": "drb",
+            "owner": "drb",
+            "state": "Uploading",
+            "transition": "1970-01-01T00:42:00+00:00",
+            "uploaded": "2022-01-01T00:00:00+00:00",
+        }
+
+    def test_dataset_keys(self, provide_metadata):
+        ds = Dataset.query(name="drb")
+        metadata = Metadata.getvalue(ds, "dataset.state")
+        assert metadata == "Uploading"
+        metadata = Metadata.getvalue(ds, "dataset.transition")
+        assert metadata == "1970-01-01T00:42:00+00:00"
+        metadata = Metadata.getvalue(ds, "dataset.nosuchkey")
+        assert metadata is None
+
+    def test_server_full(self, provide_metadata):
+        ds = Dataset.query(name="drb")
+        metadata = Metadata.getvalue(ds, "server")
+        assert metadata == {
+            "deletion": "2022-12-25",
+            "index-map": {
+                "unit-test.v6.run-data.2020-08": ["random_md5_string1"],
+                "unit-test.v5.result-data-sample.2020-08": ["random_document_uuid"],
+                "unit-test.v6.run-toc.2020-05": ["random_md5_string1"],
+            },
+        }
+
+    def test_server_keys(self, provide_metadata):
+        ds = Dataset.query(name="drb")
+        metadata = Metadata.getvalue(ds, "server.deletion")
+        assert metadata == "2022-12-25"
+        metadata = Metadata.getvalue(ds, "server.index-map")
+        assert metadata == {
+            "unit-test.v6.run-data.2020-08": ["random_md5_string1"],
+            "unit-test.v5.result-data-sample.2020-08": ["random_document_uuid"],
+            "unit-test.v6.run-toc.2020-05": ["random_md5_string1"],
+        }
```

Comment:
No need for a `server.nosuchkey` test?

Reply:
Not really, since there's no unique logic here; but it can't hurt.